include/odp/api/spec/ipsec.h

```diff
@@ -1224,16 +1224,22 @@ typedef struct odp_ipsec_status_t {
  *
  * The operation does packet transformation according to IPSEC standards (see
  * e.g. RFC 4302 and 4303). Resulting packets are well formed, reconstructed
- * original IP packets, with IPSEC headers removed and valid header field values
- * restored. The amount and content of packet data before the IP header is
- * undefined.
+ * original IP (or TFC dummy) packets, with IPSEC headers removed and valid
+ * header field values restored. The amount and content of packet data before
+ * the IP header is undefined. TFC padding may follow the IP packet payload,
+ * in which case packet length is larger than protocol headers indicate.
+ * TFC dummy packets in tunnel mode have both IPv4 and IPv6 flags cleared and
+ * may contain some (dummy) data.
  *
  * Each successfully transformed packet has a valid value for these metadata
  * regardless of the inner packet parse configuration
  * (odp_ipsec_inbound_config_t):
- * - L3 offset: Offset to the first byte of the (outmost) IP header
- * - pktio:     For inline IPSEC processed packets, original packet input
- *              interface
+ * - l3_offset:  Offset to the first byte of the original IP packet. The value
+ *               is implementation specific for tunnel mode TFC dummy packets.
+ * - has_ipv4/6: Specifies if the original packet is IPv4 or IPv6. For tunnel
+ *               mode TFC dummy packets neither flag is set.
```

Comment:
In tunnel mode not setting ipv4/v6 flag for dummy packet should be fine. But how do Dummy packets get handled in transport mode? coz if the implementation does inner packet parsing then transport mode dummy packet will have the valid ipv4/v6 flag set.

Reply:
Transport mode dummy packets _are_ IP packets. The protocol field of the IP header (which is constructed based on the IP header of the ESP packet) is 59.

Reply:
Yes. Just to be clear even if the implementation know that its a dummy packet in Transport mode the application/implementation will have to parse the IP header to know the dummy packet and drop the packet.